test/local/account_routes.js

```diff
@@ -507,13 +548,32 @@ test('/account/login', function (t) {
       t.equal(mockDB.createSessionToken.callCount, 1, 'db.createSessionToken was called')
       t.equal(mockDB.sessions.callCount, 1, 'db.sessions was called')
 
-      t.equal(mockLog.stdout.write.callCount, 1, 'an sqs event was logged')
+      t.equal(mockDB.createDevice.callCount, 1, 'db.createDevice was called once')
+      var args = mockDB.createDevice.args[0]
+      t.deepEqual(args[0], uid, 'first argument was uid')
+      t.deepEqual(args[1], tokenId, 'second argument was tokenId')
+      t.deepEqual(args[2], {}, 'third argument was empty object')
+
+      t.equal(mockPush.notifyDeviceConnected.callCount, 1, 'push.notifyDeviceConnected was called once')
+      args = mockPush.notifyDeviceConnected.args[0]
+      t.deepEqual(args[0], uid, 'first argument was uid')
+      t.equal(args[1], undefined, 'second argument was undefined')
+      t.equal(args[2], deviceId.toString('hex'), 'third argument was stringified deviceId')
+
+      t.equal(mockLog.stdout.write.callCount, 2, 'two events were logged')
+
       var eventData = JSON.parse(mockLog.stdout.write.getCall(0).args[0])
-      t.equal(eventData.event, 'login', 'it was a login event')
-      t.equal(eventData.data.service, 'sync', 'it was for sync')
-      t.equal(eventData.data.email, TEST_EMAIL, 'it was for the correct email')
-      t.equal(eventData.data.metricsContext.entrypoint, 'preferences', 'it contained the entrypoint metrics field')
-      t.equal(eventData.data.metricsContext.utm_content, 'some-content-string', 'it contained the utm_content metrics field')
+      t.equal(eventData.event, 'device:create', 'first event was a device:create event')
+      t.equal(eventData.data.id, deviceId.toString('hex'), 'first event had correct id')
+      t.equal(eventData.data.timestamp, deviceCreatedAt, 'first event had correct timestamp')
+      t.equal(eventData.data.type, undefined, 'first event had no type')
+
+      eventData = JSON.parse(mockLog.stdout.write.getCall(1).args[0])
+      t.equal(eventData.event, 'login', 'second event was a login event')
+      t.equal(eventData.data.service, 'sync', 'second event was for sync')
+      t.equal(eventData.data.email, TEST_EMAIL, 'second event was for the correct email')
+      t.equal(eventData.data.metricsContext.entrypoint, 'preferences', 'second event contained the entrypoint metrics field')
+      t.equal(eventData.data.metricsContext.utm_content, 'some-content-string', 'second event contained the utm_content metrics field')
```

Comment:
Fwiw, because this involved changing the position of the `login` event, which was otherwise unrelated to this PR, I've done this in a separate commit: 87df019.

Why did I move the `login` event rather than the call to `devices.upsert`?

Because device creation was already being invoked at a consistent juncture for both the `/account/create` and `/account/login` endpoints, immediately after the session token is created. The `login` event, on the other hand, was being emitted before the session token is created for `/account/create` but after the key fetch token is created for `/account/login`. So, to avoid making two things inconsistent, I moved the event rather than the device creation. I hope that's cool.